### PR TITLE
Yarn

### PR DESCRIPTION
**Pull request by WhenWen**
Implemented 'yet another rope embedding' from https://arxiv.org/abs/2309.00071.

The hyperparameter follows the same meaning in huggingface implementation https://github.com/huggingface/transformers/blob/ccf2ca162e33f381e454cdb74bf4b41a51ab976d/src/transformers/modeling_rope_utils.py#L226

### CODE REVIEW

File: src/levanter/layers/rotary.py [hunk 1]
```diff
@@ -180,6 +180,93 @@ def to_hf_config(self) -> tuple[float, dict]:
...
+            half_dim = self.HeadDim.size // 2
+            head_dim = self.HeadDim.size
+            HeadHalfSize = self.HeadDim.resize(self.HeadDim.size // 2)
+            inv_freq: NamedArray = 1.0 / (
+                self.config.theta ** (hax.arange(HeadHalfSize, step=2, dtype=jnp.float32) / head_dim)
+            )
+
+            # YaRN β-ramp
+            def _find_dim(n_rot: float):
+                return (half_dim * math.log(self.config.original_max_position_embeddings / (n_rot * 2 * math.pi))) / (
```
Comment (Helw150) [inline src/levanter/layers/rotary.py:200]: Why use `math` here rather than `jnp`? Is there a compile time advantage I'm missing?

**Review by Helw150 — APPROVED** [review]
LGTM, though I'd have a preference to use jnp.pi, etc. rather than have the math import unless I'm missing something that makes this nicer for Jit.

**Review by dlwh — APPROVED** [review]

File: src/levanter/layers/rotary.py (hunk 1 above)
Comment (WhenWen) [reply src/levanter/layers/rotary.py:200]: Have some bug initially with jnp. Will keep it this way now
